File: 01/play.js
function replaceStringWithNumbers(text_line) {
    tempString = text_line
    const numbers = {
        zero: "z0o",
        one: "o1e",
        two: "t2o",
        three:"t3e",
        four: "f4r",
        five: "f5e",
        six: "s6x",
        seven: "s7n",
        eight: "e8t",
        nine: "n9e"
    }

    Object.keys(numbers).forEach (key => {
        // console.log(key, numbers[key]);
        tempString = tempString.replaceAll(key, numbers[key])
    })

    return tempString
}

test = "eightwothree";




console.log(test)
// console.log(numbers.one)
console.log(replaceStringWithNumbers(test))

File: 01/play.dart
import 'dart:io';

String nameToNum(String line){
    String tempString = line;
    Map<String, String> digits = {
        "zero": "z0o",
        "one": "o1e",
        "two": "t2o",
        "three":"t3e",
        "four": "f4r",
        "five": "f5e",
        "six": "s6x",
        "seven": "s7n",
        "eight": "e8t",
        "nine": "n9e"
    };
    digits.forEach((k, v) {
        tempString = tempString.replaceAll(k,v);
    });
    return tempString;
}

String getFirstNumber(String line){
    int indexOfFirstNum;

    indexOfFirstNum = line.indexOf(RegExp(r'\d'));
    return line[indexOfFirstNum];
}

void main() async {

    String test = "eightwothree";
    print(test);
    print(nameToNum(test));

}